### PR TITLE
ci: enable more Firefox tests that are now passing and fix others

### PR DESCRIPTION
**Pull request by whimboo**

### CODE REVIEW

File: test/src/requestinterception-experimental.spec.ts [hunk 1]
```diff
@@ -341,9 +341,13 @@ describe('cooperative request interception', function () {
...
       expect(failedRequest).toBeTruthy();
-      expect(failedRequest.failure()!.errorText).toBe(
-        'net::ERR_INTERNET_DISCONNECTED'
-      );
+      if (isChrome) {
+        expect(failedRequest.failure()!.errorText).toBe(
+          'net::ERR_INTERNET_DISCONNECTED'
+        );
+      } else {
+        expect(failedRequest.failure()!.errorText).toBe('NS_ABORT_ERROR');
```
Comment (Lightning00Blade) [inline test/src/requestinterception-experimental.spec.ts]: This is wrong as BiDi does not support custom error like the test tries to test.

File: test/src/requestinterception.spec.ts [hunk 2]
```diff
@@ -258,9 +258,13 @@ describe('request interception', function () {
       ]);
 
       expect(failedRequest).toBeTruthy();
-      expect(failedRequest.failure()!.errorText).toBe(
-        'net::ERR_INTERNET_DISCONNECTED'
-      );
+      if (isChrome) {
+        expect(failedRequest.failure()!.errorText).toBe(
+          'net::ERR_INTERNET_DISCONNECTED'
```
Comment (Lightning00Blade) [inline test/src/requestinterception.spec.ts]: Same as above.

File: test/TestExpectations.json [hunk 3]
```diff
@@ -3315,33 +3315,19 @@
...
-    "expectations": ["SKIP"],
-    "comment": "TODO: Needs full support for continueRequest in Firefox https://bugzilla.mozilla.org/show_bug.cgi?id=1850680"
-  },
   {
     "testIdPattern": "[requestinterception-experimental.spec] cooperative request interception Page.setRequestInterception should cooperatively respond by priority",
     "platforms": ["darwin", "linux", "win32"],
     "parameters": ["firefox", "webDriverBiDi"],
     "expectations": ["SKIP"],
-    "comment": "TODO: Needs full support for continueRequest in Firefox https://bugzilla.mozilla.org/show_bug.cgi?id=1850680"
+    "comment": "TODO: Needs full support for continueResponse in Firefox https://bugzilla.mozilla.org/show_bug.cgi?id=1849104"
```
Comment (Lightning00Blade) [inline test/TestExpectations.json]: We don't use that command for Puppeteer we use `network.provideResponse`.

**Review by Lightning00Blade — CHANGES_REQUESTED** [review]

File: test/src/requestinterception-experimental.spec.ts (hunk 1 above)
Comment (Lightning00Blade) [reply test/src/requestinterception-experimental.spec.ts]: WebDriver BiDi issue: https://github.com/w3c/webdriver-bidi/issues/508

Comment (whimboo) [reply test/src/requestinterception-experimental.spec.ts]: So should this test be skipped for both chrome and firefox, and ignored for now?

File: test/TestExpectations.json (hunk 3 above)
Comment (whimboo) [reply test/TestExpectations.json]: Thanks. Will update the name then.

File: test/src/requestinterception-experimental.spec.ts (hunk 1 above)
Comment (Lightning00Blade) [reply test/src/requestinterception-experimental.spec.ts]: The test only fail on assertion so I think it's fine to mark them as `FAIL` and update the comment to something like: 
`BiDi does not support custom errors - https://github.com/w3c/webdriver-bidi/issues/508`

File: test/src/navigation.spec.ts [hunk 4]
```diff
@@ -719,7 +719,7 @@ describe('navigation', function () {
           waitEvent(page, 'frameattached').then(_frame => {
             return (frame = _frame);
           }),
-          waitEvent(page, 'framenavigated', f => {
+          waitEvent(page, 'framenavigated').then(f => {
```
Comment (Lightning00Blade) [inline test/src/navigation.spec.ts]: Why was this changed?
This now does not wait for the correct frame to be navigated but any frame to be navigated.

File: test/src/network.spec.ts [hunk 5]
```diff
@@ -555,7 +555,7 @@ describe('network', function () {
       if (isChrome) {
         expect(failedRequest.failure()!.errorText).toBe('net::ERR_FAILED');
       } else {
-        expect(failedRequest.failure()!.errorText).toBe('NS_ERROR_FAILURE');
+        expect(failedRequest.failure()!.errorText).toBe('NS_ERROR_ABORT');
```
Comment (OrKoN) [inline test/src/network.spec.ts:558]: Interesting, so CDP in Firefox sends a different code? I wonder if this would fail for the CDP-Firefox tests.

**Review by OrKoN — COMMENTED** [review]
looks like CDP-Firefox tests are failing

File: test/src/navigation.spec.ts (hunk 4 above)
Comment (whimboo) [reply test/src/navigation.spec.ts]: Oh, this extra argument acts like a filter. I wasn't aware of that. So I'll revert this part and check why the test is failing.

File: test/src/network.spec.ts (hunk 5 above)
Comment (whimboo) [reply test/src/network.spec.ts:558]: The CDP implementation fails here anyway due to other issues.

**Review by Lightning00Blade — APPROVED** [review]

**Comment by Lightning00Blade** [comment]
Thanks!
